yaml/shared/src/main/scala/zio/config/yaml/YamlConfigSource.scala

```diff
@@ -104,20 +133,44 @@ object YamlConfigSource {
   def fromYamlString(
     yamlString: String,
     sourceName: String = "yaml"
+  ): Either[ReadError[String], ConfigSource] =
+    loadFromPropertyTree(
+      convertYaml(
+        loadYaml(yamlString)
+      ),
+      sourceName
+    )
+
+  private def loadFromPropertyTree(
+    propertyTree: => PropertyTree[String, String],
+    sourceName: String
   ): Either[ReadError[String], ConfigSource] =
     Try {
       ConfigSource.fromPropertyTree(
-        convertYaml(
-          new Load(LoadSettings.builder().build()).loadFromString(yamlString)
-        ),
+        propertyTree,
         sourceName,
         LeafForSequence.Invalid
       )
     }.toEither.swap
       .map(throwable =>
         ReadError.SourceError(
-          message = s"Failed to retrieve a valid from the yaml source. ${throwable.getMessage}"
-        ): ReadError[String]
+          message = s"Failed to retrieve a valid yaml from the yaml source. ${throwable.getMessage}",
+          Set.empty
```

Comment:
I am getting the following stacktrace in case the source is not valid 

```
Fiber failed.
An unchecked error was produced.
java.lang.NoSuchMethodError: 'scala.collection.immutable.Set zio.config.ReadError$SourceError$.apply$default$2()'
	at zio.config.yaml.YamlConfigSource$.$anonfun$loadFromPropertyTree$2(YamlConfigSource.scala:156)
YamlConfigSource.scala:156
	at scala.util.Either.map(Either.scala:382)
Either.scala:382
```

Passing this default value (empty set) changes the error to what the user expects
```
Fiber failed.
A checked error was not handled.
ReadError:
SourceError: Failed to retrieve a valid yaml from the yaml source. ErrorStringFromYamlResource
```

Reply:
Fantastic